### DISCUSSION TITLE
symfony cache:warmup

### DISCUSSION THREAD

**Discussion by olivier-sabban** [brefphp/bref#1351] [Support] [answered]
Could you please help me to understand how correctly warmup my symfony cache ?

Bref documentation tell us to warmup cache before deploying. 
https://bref.sh/docs/frameworks/symfony.html#deploy
`php bin/console cache:warmup --env=prod`

This command will generate var/cache/prod/ directory
But in lambda environnement, /tmp/ directory will be empty and BrefKernel will overload cache directory by '/tmp/cache/prod/' 
https://github.com/brefphp/symfony-bridge/blob/448540517e67a231c8281c2ce05d5409ae1fbeb1/src/BrefKernel.php#L20

Even if I submit my var/cache/prod directory in serverless artifact I still have 5sec TTFB (only on cold start - symfony hello world)  

My questions : 
- How brefphp/symfony-bridge will manage the cache directory in lambda env ? Is it still able to read var/cache/prod with BrefKernel ? 
- How can I be sure to prewarmed my cache correctly  and it won't be recreate on startup ?
- I can I decrease thoses 5 secondes
- I am not writing to a filesystem cache. You recommand to 2 methods https://github.com/brefphp/symfony-bridge#optimize-cache or psr6-dynamo-db-bundle and https://github.com/brefphp/symfony-bridge#optimize-cache. I wonder if I should try to implement them to speed up my startup. 

Many blogs discuss about this topic but they don't go in depth and just overview the solution. 
They do not specify which version of Bref and Symfony they are using..

The documentation on this subject is insufficient. It would be very nice to have a dedicated section here : https://bref.sh/docs/frameworks/symfony.html#using-cache 

By the way, thank you for this fantastic project. 
After 2 years it's still magic to launch PHP SF in AWS Lambda !
Thank you in advance for your help

**Comment by t-richard** [comment, marked as answer]
Hi,

When using the Symfony Bridge and extending from the BrefKernel, your cache folder from `var/cache/prod` should be copied to `/tmp` and this should greatly reduce the cold start time you are seeing.

If that is the case, you should see in your logs
> Symfony cache directory prepared in x ms.

If you don't see this message or it seems to not be working as expected, here are some possible reasons

* You don't extend from the BrefKernel in your Symfony Kernel 
* You don't have `APP_ENV=prod` inside lambda
* Your deployment artifact doesn't contain the var/cache folder (you can verify this by downloading the artifact from the AWS console)

Happy to help further if it doesn't lead anywhere 🙂
